core/tracker/tracker.go

```diff
@@ -559,105 +636,131 @@ func (t *Tracker) ConsensusEvent(ctx context.Context, duty core.Duty, data core.
+		parSig := parSig // Copy loop iteration values
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-t.quit:
 			return nil
 		case t.input <- event{
-			duty:      duty,
-			component: parSigEx,
-			pubkey:    pubkey,
-			shareIdx:  pSig.ShareIdx,
+			duty:   duty,
+			step:   parSigEx,
+			pubkey: pubkey,
+			parSig: &parSig,
 		}:
 		}
 	}
 
 	return nil
 }
 
-// ParSigDBInternalEvent inputs events from core.ParSigDB component for internal store event.
+// ParSigDBInternalEvent inputs events from core.ParSigDB step event for local VC submitted parsigs.
 func (t *Tracker) ParSigDBInternalEvent(ctx context.Context, duty core.Duty, data core.ParSignedDataSet) error {
-	for pubkey, pSig := range data {
+	for pubkey, parSig := range data {
+		parSig := parSig // Copy loop iteration values
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-t.quit:
 			return nil
 		case t.input <- event{
-			duty:      duty,
-			component: parSigDBInternal,
-			pubkey:    pubkey,
-			shareIdx:  pSig.ShareIdx,
+			duty:   duty,
+			step:   parSigDBInternal,
+			pubkey: pubkey,
+			parSig: &parSig,
 		}:
 		}
 	}
 
 	return nil
 }
 
-// ParSigDBThresholdEvent inputs event from core.ParSigDB component for threshold event.
+// ParSigDBThresholdEvent inputs event from core.ParSigDB step for threshold emitted parsigs.
 func (t *Tracker) ParSigDBThresholdEvent(ctx context.Context, duty core.Duty, pubkey core.PubKey, _ []core.ParSignedData) error {
 	select {
 	case <-ctx.Done():
 		return ctx.Err()
 	case <-t.quit:
 		return nil
 	case t.input <- event{
-		duty:      duty,
-		component: parSigDBThreshold,
-		pubkey:    pubkey,
+		duty:   duty,
+		step:   parSigDBThreshold,
+		pubkey: pubkey,
 	}:
 	}
 
 	return nil
 }
 
-// SigAggEvent inputs event from core.SigAgg component.
+// SigAggEvent inputs event from core.SigAgg step.
 func (t *Tracker) SigAggEvent(ctx context.Context, duty core.Duty, pubkey core.PubKey, _ core.SignedData) error {
 	select {
 	case <-ctx.Done():
 		return ctx.Err()
 	case <-t.quit:
 		return nil
 	case t.input <- event{
-		duty:      duty,
-		component: sigAgg,
-		pubkey:    pubkey,
+		duty:   duty,
+		step:   sigAgg,
+		pubkey: pubkey,
 	}:
 	}
 
 	return nil
 }
+
+func parSigReporter(ctx context.Context, duty core.Duty, parsigMsgs parsigsByMsg) {
+	if len(parsigMsgs) <= 1 {
+		return // Nothing to report.
+	}
+
+	inconsistentCounter.WithLabelValues(duty.Type.String()).Inc()
+
+	if expectInconsistentParSigs(duty.Type) {
+		log.Debug(ctx, "Inconsistent sync committee partial signed data",
+			z.Any("data", parsigMsgs))
+	} else {
+		log.Warn(ctx, "Inconsistent partial signed data", nil,
+			z.Any("data", parsigMsgs))
+	}
+}
+
+// expectInconsistentParSigs returns true if the duty type is expected to sometimes
+// produce inconsistent partial signed data.
+func expectInconsistentParSigs(duty core.DutyType) bool {
+	return duty == core.DutySyncMessage || duty == core.DutySyncContribution
```

Comment:
shouldn't this be
```suggestion
	return duty == core.DutySyncMessage || duty == core.DutyPrepareSyncContribution
```
??

Reply:
no, `SyncCommitteeSelection` should always be consistent, whlie `ContributionAndProof` contains the beaconblockroot that VC pick themselves.

Reply:
true